components/ee/ws-scheduler/pkg/scheduler/state.go

```diff
@@ -112,33 +120,51 @@ func ComputeState(nodes []*corev1.Node, pods []*corev1.Pod, bindings []*Binding,
 			pds[b.Pod.Name] = b.Pod
 		}
 
-		podToNode[b.Pod.Name] = b.NodeName
+		podToNode[b.Pod.Name] = &podAndNode{
+			pod:      b.Pod,
+			nodeName: b.NodeName,
+		}
 	}
 
 	// With a unique pod to node assignment, we can invert that relationship and compute
 	// which node has which pods. If we did this right away, we might assign the same pod
 	// to multiple nodes.
-	nodeToPod := make(map[string]map[string]struct{}, len(nodes))
+	type ntp struct {
+		pods   map[string]struct{}
+		ghosts map[string]struct{}
+	}
+	nodeToPod := make(map[string]*ntp, len(nodes))
 	for _, n := range nodes {
 		nds[n.Name] = &Node{
 			Node:     n,
 			Services: make(map[string]struct{}),
 		}
-		nodeToPod[n.Name] = make(map[string]struct{})
+		nodeToPod[n.Name] = &ntp{
```

Comment:
why not make this part of `nds`?

Reply:
Good question - I left it as I found it. :upside_down_face: 
But the reasons seems to be that the book keeping is done during the process of adding pods from `pds` to each `node` in `nds` [here](https://github.com/gitpod-io/gitpod/blob/9b7a8941ad0ec204bb12fe5e7b32457c6af51c0a/components/ee/ws-scheduler/pkg/scheduler/state.go#L172). So it make sense to keep them separate.